### DISCUSSION TITLE
I need to reinstall the app in the middle of the session

### DISCUSSION THREAD

**Discussion by LizGrillo** [appium/appium#19738] [Q&A] [answered]
**Preconditions**: I already checked here: https://discuss.appium.io/t/differences-between-activateapp-terminateapp-closeapp-launchapp-driver-methods/40371/2
and here: https://discuss.appium.io/t/will-it-be-possible-to-uninstall-and-reinstall-the-app-in-a-single-session-on-both-ios-and-android/31585

Now that we don't have the close() and launch() methods anymore, I don't know very well how to delete the app and install it again between @tests.

Im using this:

@AfterMethod(onlyForGroups = {"group1","group2"})
    public void deletingAccountAndReinstallingApp() {
        login.completeDeleteAccountProcess();
        driver.terminateApp(bundleID);
        driver.removeApp("bundleID");
        driver.installApp(path-to-apk);
        driver.activateApp("bundleID");
    }

Is there a way to do this better? 'cause sometimes the session falls and everything fails because it says the bundle id is still up.

Please, if someone can help here, could you please add detailed steps, since Im not a senior.

**Comment by KazuCocoa** [comment]
Uninstalling the app kills the bundle ID's process as well, so if you did not need to kill the app process explicitly, `driver.terminateApp(bundleID);` is redundant.

**Reply by LizGrillo** [reply]
Ok, deleted, thanks. 

There's no other way to do this?

**Reply by KazuCocoa** [reply, marked as answer]
No, basically these steps are what the deprecated `launch()` method did to re-install the app. In case you'd like to start the app process with process arguments, https://appium.github.io/appium-xcuitest-driver/5.15/reference/execute-methods/#mobile-launchapp is the alternative of `activateApp`.

**Reply by LizGrillo** [reply]
Thanks @KazuCocoa ! I'll take a look!
